doc/developer-guides/hld/hv-vcat.rst

```diff
@@ -0,0 +1,141 @@
+shared cache ways between low priority VMs.
+
+In ACRN, vCAT can be configured via the "VM-Configuration". The vCAT cache ways allocated
+for VMs are determined in the VM configuration (:ref:`vcat_configuration`).
+
+For further details on the RDT, refer to the ACRN RDT high-level design :ref:`hv_rdt`.
+
+
+High Level ACRN vCAT Design
+***************************
+
+ACRN CAT virtualization support can be divided into two parts:
+
+- CAT Capability Exposure to Guest
+
+- CAT resources (CLOS IDs + cache ways) management
+
+The figure below shows high-level design of vCAT in ACRN:
+
+   .. figure:: images/vcat-hld.png
+      :align: center
+
+CAT Capability Exposure to Guest
+********************************
+ACRN exposes CAT capability and resource to a guest VM via vCPUID and vMSR, as explained
+in the following sections.
+
+vCPUID
+======
+
+CPUID Leaf 07H
+--------------
+
+- CPUID.(EAX=07H, ECX=0).EBX.PQE[bit 15]: Supports RDT capability if 1. This bit will be set for a vCAT VM.
+
+CPUID Leaf 10H
+--------------
+
+**CAT Resource Type and Capability Enumeration**
+
+- CPUID.(EAX=10H, ECX=0):EBX[1]: If 1, indicate L3 CAT support for a vCAT VM.
+- CPUID.(EAX=10H, ECX=0):EBX[2]: If 1, indicate L2 CAT support for a vCAT VM.
+- CPUID.(EAX=10H, ECX=1): CAT capability enumeration sub-leaf for L3. Reports L3 COS_MAX and CBM_LEN to a vCAT VM
+- CPUID.(EAX=10H, ECX=2): CAT capability enumeration sub-leaf for L2. Reports L2 COS_MAX and CBM_LEN to a vCAT VM
+
+vMSR
+====
+
+The following CAT MSRs will be virtualized for a vCAT VM:
+
+- IA32_PQR_ASSOC
+- IA32_type_MASK_0 ~ IA32_type_MASK_n
+
+By default, after reset, all CPU cores are assigned to COS 0 and all IA32_type_MASK_n MSRs
+are programmed to allow fill into all cache ways.
+
+
+CAT resources (CLOS IDs + cache ways) management
+************************************************
+
+All accesses to the CAT MSRs are intercepted by vMSR and control is passed to vCAT, which will perform
+the following actions:
+
+- Intercept IA32_PQR_ASSOC MSR to re-map virtual CLOS ID to physical CLOS ID.
+  Upon writes, store the re-mapped physical CLOS ID into its vCPU ``msr_store_area``
+  data structure guest part. It will be loaded to physical IA32_PQR_ASSOC on each VM-Enter.
+
+
+- Intercept IA32_type_MASK_n MSRs to re-map virtual CBM to physical CBM. Upon writes,
+  program re-mapped physica CBM into corresponding physical IA32_type_MASK_n MSR
+
+  Several vCAT P2V (physical to virtual) and V2P (virtual to physical)
+  mappings exist, as illustrated in the following pseudocode:
+
+.. code-block:: none
+
+       struct acrn_vm_config *vm_config = get_vm_config(vm_id)
+
+       max_pcbm = vm_config->max_type_pcbm (type: l2 or l3)
+       mask_shift = ffs64(max_pcbm)
+
+       vclosid = vmsr - MSR_IA32_type_MASK_0
+       pclosid = vm_config->pclosids[vclosid]
+
+       pmsr = MSR_IA32_type_MASK_0 + pclosid
+       pcbm = vcbm << mask_shift
+       vcbm = pcbm >> mask_shift
+
+       Where
```

Comment:
We should take this part out of the `code-block`, and use double-backticks to highlight the the code-related items.